Domain/SymfonyAclCache.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Acl\Domain;
+
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\Security\Acl\Model\AclCacheInterface;
+use Symfony\Component\Security\Acl\Model\AclInterface;
+use Symfony\Component\Security\Acl\Model\ObjectIdentityInterface;
+use Symfony\Component\Security\Acl\Model\PermissionGrantingStrategyInterface;
+
+/**
+ * This class is a wrapper around a Symfony cache implementation.
+ *
+ * @author Michael Babker <michael.babker@gmail.com>
+ */
+class SymfonyAclCache implements AclCacheInterface
+{
+    const PREFIX = 'sf_acl_';
+
+    private $cache;
+    private $prefix;
+    private $permissionGrantingStrategy;
+
+    /**
+     * Constructor.
+     *
+     * @param string $prefix
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(AdapterInterface $cache, PermissionGrantingStrategyInterface $permissionGrantingStrategy, $prefix = self::PREFIX)
+    {
+        if (0 === \strlen($prefix)) {
+            throw new \InvalidArgumentException('$prefix cannot be empty.');
+        }
+
+        $this->cache = $cache;
+        $this->permissionGrantingStrategy = $permissionGrantingStrategy;
+        $this->prefix = $prefix;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clearCache()
+    {
+        $this->cache->clear($this->prefix);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function evictFromCacheById($aclId)
+    {
+        $lookupKey = $this->getAliasKeyForIdentity($aclId);
+        if (!$this->cache->hasItem($lookupKey)) {
+            return;
+        }
+
+        $cacheItem = $this->cache->getItem($lookupKey);
+        if ($cacheItem->isHit()) {
+            $key = $cacheItem->get();
+            if ($this->cache->hasItem($key)) {
+                $this->cache->deleteItem($key);
+            }
+        }
+
+        $this->cache->deleteItem($lookupKey);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function evictFromCacheByIdentity(ObjectIdentityInterface $oid)
+    {
+        $key = $this->getDataKeyByIdentity($oid);
+        if (!$this->cache->hasItem($key)) {
+            return;
+        }
+
+        $this->cache->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFromCacheById($aclId)
```

Comment:
```suggestion
    public function getFromCacheById($aclId): ?AclInterface
```

This is a new class. We should add return types.

Reply:
Typehints in general are added on the new class